applications/FluidDynamicsApplication/python_scripts/navier_stokes_two_fluids_solver.py

```diff
@@ -0,0 +1,269 @@
+from __future__ import print_function, absolute_import, division  # makes KratosMultiphysics backward compatible with python 2.6 and 2.7
+
+# Importing the Kratos Library
+import KratosMultiphysics
+
+# Check that applications were imported in the main script
+KratosMultiphysics.CheckRegisteredApplications("FluidDynamicsApplication")
+
+# Import applications
+import KratosMultiphysics.FluidDynamicsApplication as KratosCFD
+
+# Import base class file
+from fluid_solver import FluidSolver
+
+# from KratosMultiphysics.FluidDynamicsApplication import *
+
+
+
+def CreateSolver(model, custom_settings):
+    return NavierStokesTwoFluidsSolver(model, custom_settings)
+
+class NavierStokesTwoFluidsSolver(FluidSolver):
+
+    def _ValidateSettings(self, settings):
+        ##settings string in json format
+        default_settings = KratosMultiphysics.Parameters("""
+        {
+            "solver_type": "two_fluids_solver_from_defaults",
+            "model_part_name": "",
+            "domain_size": -1,
+            "model_import_settings": {
+                "input_type": "mdpa",
+                "input_filename": "unknown_name",
+                "reorder": false
+            },
+            "distance_reading_settings"    : {
+                "import_mode"         : "from_mdpa",
+                "distance_file_name"  : "no_distance_file"
+            },
+            "maximum_iterations": 7,
+            "dynamic_tau": 1.0,
+            "echo_level": 0,
+            "time_order": 2,
+            "compute_reactions": false,
+            "reform_dofs_at_each_step": false,
+            "relative_velocity_tolerance": 1e-3,
+            "absolute_velocity_tolerance": 1e-5,
+            "relative_pressure_tolerance": 1e-3,
+            "absolute_pressure_tolerance": 1e-5,
+            "linear_solver_settings"       : {
+                "solver_type"         : "AMGCL"
+            },
+            "volume_model_part_name" : "volume_model_part",
+            "skin_parts": [""],
+            "no_skin_parts":[""],
+            "time_stepping"                : {
+                "automatic_time_step" : true,
+                "CFL_number"          : 1,
+                "minimum_delta_time"  : 1e-2,
+                "maximum_delta_time"  : 1.0
+            },
+            "periodic": "periodic",
+            "move_mesh_flag": false
+        }""")
+
+        settings.ValidateAndAssignDefaults(default_settings)
+        return settings
+
+    def __init__(self, model, custom_settings):
+        super(NavierStokesTwoFluidsSolver,self).__init__(model,custom_settings)
+
+        self.element_name = "TwoFluidNavierStokes"
+        self.condition_name = "NavierStokesWallCondition"
+        
+        self.min_buffer_size = 3
+
+        # There is only a single rank in OpenMP, we always print
+        self._is_printing_rank = True
+
+        ## Construct the linear solver
+        import linear_solver_factory
```

Comment:
I think doing this in the constructor can make problems if you want to derive a trilinos version

Reply:
was removed

Reply:
( now in Initialize() )